lib/cusolver/dense_generic.jl

```diff
@@ -286,29 +286,15 @@ function Xgesvdp!(jobz::Char, econ::Int, A::StridedCuMatrix{T}) where {T <: Blas
     m, n = size(A)
     p = min(m, n)
     R = real(T)
-    econ ∈ (0, 1) || throw(ArgumentError("econ is incorrect. The values accepted are 0 and 1."))
-    U = if jobz == 'V' && econ == 1
-        CuMatrix{T}(undef, m, p)
-    elseif jobz == 'V' && econ == 0
-        CuMatrix{T}(undef, m, m)
-    elseif jobz == 'N'
-        CU_NULL
-    else
-        throw(ArgumentError("jobz is incorrect. The values accepted are 'V' and 'N'."))
-    end
+    econ == 0 || econ == 1 || throw(ArgumentError("econ is incorrect. The values accepted are 0 and 1."))
+    jobz == 'N' || jobz == 'V' || throw(ArgumentError("jobz is incorrect. The values accepted are 'V' and 'N'."))
+
+    U = (jobz == 'V' && econ == 0) ? CuMatrix{T}(undef, m, m) : CuMatrix{T}(undef, m, p)
```

Comment:
I personally prefer simple `if` statements over these IMO hard-to-read ternary expressions.

Wouldn't it also be safer to use the full `(m,p)` version for the `N` case until we hear back from NVIDIA (i.e., only switch based on `econ`)?

Reply:
Yes, it is safer. We don't know how they use `U` and `V` when `jobz = 'N'` so we should take any risk.